app/src/main/java/com/babylon/wallet/android/presentation/transfer/TransferViewModel.kt

```diff
@@ -477,6 +472,22 @@ sealed class TargetAccount {
     ) : TargetAccount() {
         override val address: String
             get() = account.address
+
+        fun hasAnyDenyDepositRule(forSpecificAssetAddress: String): Boolean {
+            val thirdPartyDeposits = this.account.onLedgerSettings.thirdPartyDeposits
+
+            val hasDenyAll = thirdPartyDeposits.depositRule == ThirdPartyDeposits.DepositRule.DenyAll
+            val hasAnyDenyExceptionRule = thirdPartyDeposits.assetsExceptionList.any {
+                it.exceptionRule == Network.Account.OnLedgerSettings.ThirdPartyDeposits.DepositAddressExceptionRule.Deny &&
+                    it.address == forSpecificAssetAddress
+            }
+
+            if (hasDenyAll || hasAnyDenyExceptionRule) {
```

Comment:
Just checking.. What instruction should be use if `hasDenyAll` is false (there is no **DenyAll** rule) but it has one deny exception rule ? This method would return false so try_deposit_or_abort I guess, but is it expected ? 🤔

Reply:
what do you mean if it is expected? 👀 

The logic is that the `hasAnyDenyExceptionRule` is true only when the specific asset we want to transfer has deny rule. => Then the deposit will be used.

Otherwise it will return false => then the try_deposit will be used.

Reply:
renamed the `hasAnyDenyExceptionRule` to make it more clear

Reply:
I see okay just wanted to make sure that having deny exception rule is enough condition to use deposit (without having denyAll) 👍